pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/errorprone/CloseResourceRule.java

```diff
@@ -121,7 +121,7 @@ private void checkForResources(Node node, Object data) {
         for (ASTLocalVariableDeclaration var : vars) {
             ASTType type = var.getTypeNode();
 
-            if (type.jjtGetChild(0) instanceof ASTReferenceType) {
+            if (!var.isTypeInferred() && type != null && type.jjtGetChild(0) instanceof ASTReferenceType) {
```

Comment:
the check for null and `var.isTypeInferred()` basically check the same thing here... I'd stick with the null check